.github/workflows/create_llvm.yml

```
@@ -35,23 +53,61 @@ jobs:
         with:
           path:
             llvm_install/**
-          key: llvm-${{ matrix.os }}-v${{ matrix.version }}-${{ matrix.build_type }}
+          key: llvm-${{ matrix.os }}-${{ matrix.arch }}-v${{ matrix.version }}-${{ matrix.build_type }}
+
+      - name: show arch flags
+        if: steps.cache.outputs.cache-hit != 'true'      
+        run:
+          echo ${{ matrix.arch_flags }}
 
-      - name: Checkout repo
+      - name: Checkout repo llvm
         if: steps.cache.outputs.cache-hit != 'true'
         uses: actions/checkout@v4
         with:
           repository: llvm/llvm-project
           ref: release/${{matrix.version}}.x
 
+      - name: Checkout repo ock platform
+        if: ${{ steps.cache.outputs.cache-hit != 'true' && matrix.arch != 'x86_64' }}
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            platform
+          path: ock          
+
       - name: Install Ninja
         if: steps.cache.outputs.cache-hit != 'true'
         uses: llvm/actions/install-ninja@main
 
+      - name: install aarch64 build tools
+        if: ${{ steps.cache.outputs.cache-hit != 'true' && matrix.arch == 'aarch64' }}
+        run:
+          sudo apt-get install --yes gcc-11-aarch64-linux-gnu g++-11-aarch64-linux-gnu
```

```suggestion
          sudo apt-get install --yes g++-11-aarch64-linux-gnu
```
? `gcc-11-aarch64-linux-gnu` should already be a dependency of that.

---

BTW, I'm not entirely sure whether installing packages during every run is fine, ideally we would cache this (https://github.com/marketplace/actions/cache-apt-packages) but I understand we are already hitting the cache limits.

---

I'll look into caching at some point, but at least this has consistency with what we do elsewhere. I'll try dropping one - I was just matching what we install in gitlab, but will try your suggestion.